### PR TITLE
Enable madvise by default for all builds

### PR DESCRIPTION
**Pull request by benwtrent**
This feature flag has been enabled by default for snapshot builds for some time, no significant bumps or changes in rally. 

This commit will enable it by default, even in release builds.

### CODE REVIEW

**Comment by benwtrent** [comment]
@elasticmachine update branch

File: distribution/src/config/jvm.options [hunk 1]
```diff
@@ -58,6 +58,9 @@
 # result in less optimal vector performance
 20-:--add-modules=jdk.incubator.vector
 
+# Enable random access flags via madvise
+-Des.madv_random_feature_flag_enabled=true
```
Comment (mark-vieira) [inline distribution/src/config/jvm.options:62]: I think typically when we want to enable something by default that was previously behind a feature flag, we simply remove the feature flag. In either case, I don't think we'd want this option in `jvm.options` but rather in `SystemJvmOptions`. @rjernst might provide more insight here.

Comment (rjernst) [reply distribution/src/config/jvm.options:62]: That's right, we should remove the feature flag to "enable" it permanently. If the intent is for users to be able to disable this, then it should be moved from a feature flag to a setting.

Comment (benwtrent) [reply distribution/src/config/jvm.options:62]: roger! I will just remove the flag :)

**Review by mark-vieira — APPROVED** [review]

**Review by ChrisHegarty — APPROVED** [review]
I agree with the complete removal of the feature flag. LGTM

**Comment by benwtrent** [comment]
@elasticmachine update branch

**Pull request merged**